platform/cxxsupport/TOOLCHAIN_ARMC5/tuple

```diff
@@ -0,0 +1,734 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2019 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* class tuple implementation based on Sashsa Goldstein's series
+ * "Implementing std::tuple from The Ground Up".
+ *
+ * http://blogs.microsoft.co.il/sasha/2015/01/12/implementing-tuple-part-1/
+ *
+ * tuple_cat based on Peter Dimov's article "Simple C++11 metaprogramming",
```

Comment:
Thanks for putting the references; it could help future maintainers. Dimov's article really is brilliant and easy to understand.

Reply:
Wasn't going to claim I'd devised all that myself :)

> future maintainers

This is only a bridge for as long as ARMC5 limps on, so I'm not expecting a particularly extended lifetime in Mbed OS anyway.